### PR TITLE
[JENKINS-71726] [JENKINS-71727] remove inline javascript

### PR DESCRIPTION
**Pull request by mawinter69**
See [JENKINS-71727](https://issues.jenkins.io/browse/JENKINS-71727).
See [JENKINS-71726](https://issues.jenkins.io/browse/JENKINS-71726).

### Testing done

JENKINS-71727:
- Do not log in
- visit a page that is restricted
- you get redirected to the login page
- login
- get redirected to the original page
- run `curl -u http://jenkins/job/test/` and checked output contains data-redirect-url to loginform and javascript src points to proper script url.
- Message `Authentication required` is included in the output

JENKINS-71726:
- Create a job without parameters
- run it once
- change job to require parameters
- run job with parameters
- visit job/2/parameters
- click on `Previous Build`
- you get redirected to job/1/
- run `curl -u user:token http://jenkins/job/test/1/parameters/` and checked output contains data-redirect-url and javascript src points to proper script url.
- Proper message `Not found` is included in the output

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [x] The Jira issue, if it exists, is well-described.
- [x] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [x] There is automated testing or an explanation as to why this change has no tests.
- [x] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [x] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [x] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [x] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [x] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

N/A

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [x] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [x] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [x] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [x] Proper changelog labels are set so that the changelog can be generated automatically.
- [x] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [x] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

**Comment by Kevin-CB** [comment]
Thanks for this PR @mawinter69, I'll try to review it as soon as possible.

File: core/src/main/java/hudson/Util.java [hunk 1]
```diff
@@ -1856,6 +1858,20 @@ public static long daysElapsedSince(@NonNull Date date) {
...
     }
 
+    @Restricted(NoExternalUse.class)
+    public static void printRedirect(String contextPath, String redirectUrl, String message, PrintWriter out) {
+        out.printf(
+                "<html><head>" +
+                "<meta http-equiv='refresh' content='1;url=%1$s'/>" +
+                "<script id='redirect' data-redirect-url='%1$s' src='" +
+                contextPath + Jenkins.RESOURCE_PATH +
+                "/scripts/redirect.js'/>" +
```
Comment (Kevin-CB) [inline core/src/main/java/hudson/Util.java:1868]:
```suggestion
                "/scripts/redirect.js'></script>" +
```

**Review by Kevin-CB — CHANGES_REQUESTED** [review]
I investigated further as to why I was experiencing this delay, and found that it was simply because the script wasn't being executed. The delay I was encountering was due to the redirection initiated by the meta refresh tag, which has a delay set to 1 second (`content='1`).

Upon closer look on what happened, the rendered page was: 

image

The script tag is malformed, taking the remainder of our request. 
I'm not sure why, but the self-closing tag isn't functioning correctly. Closing it with a full tag resolves this issue.

File: core/src/main/java/hudson/Util.java (hunk 1 above)
Comment (daniel-beck) [reply core/src/main/java/hudson/Util.java:1868]: > I'm not sure why

https://html.spec.whatwg.org/#elements-2

`script` is a **raw text element**, and

> [Raw text](https://html.spec.whatwg.org/#raw-text-elements) … elements have a [start tag](https://html.spec.whatwg.org/#syntax-start-tag) to indicate where they begin, and an [end tag](https://html.spec.whatwg.org/#syntax-end-tag) to indicate where they end. The start and end tags of certain [other tags] can be [omitted](https://html.spec.whatwg.org/#syntax-tag-omission) …. Those that cannot be omitted must not be omitted.

**Review by Kevin-CB — APPROVED** [review]
I retested it with your change, and it works fine.

**Review by MarkEWaite — APPROVED** [review]
I've tested with this change applied to the master branch and it  works well in my testing also.

This PR is now ready for merge.  We will merge it after approximately 24 hours if there is no negative feedback.

/label ready-for-merge

**Pull request merged**